brain/circuit.cpp

```
@@ -662,14 +679,78 @@ Matrix4fs Circuit::getTransforms( const GIDSet& gids ) const
 
 }
 
+Quaternionfs Circuit::getRotations( const GIDSet& gids ) const
+{
+    return _impl->getRotations( gids );
+}
+
 size_t Circuit::getNumNeurons() const
 {
     return _impl->getNumNeurons();
 }
 
-Quaternionfs Circuit::getRotations( const GIDSet& gids ) const
+SynapsesStream Circuit::getAfferentSynapses( const GIDSet& gids,
+                                          const SynapsePrefetch prefetch ) const
 {
-    return _impl->getRotations( gids );
+    return SynapsesStream( *this, gids, true, prefetch );
+}
+
+SynapsesStream Circuit::getEfferentSynapses( const GIDSet& gids,
+                                          const SynapsePrefetch prefetch ) const
+{
+    return SynapsesStream( *this, gids, false, prefetch );
+}
+
+SynapsesStream Circuit::getProjectedSynapses( const GIDSet& preGIDs,
+                                              const GIDSet& postGIDs,
+                                          const SynapsePrefetch prefetch ) const
+{
+    return SynapsesStream( *this, preGIDs, postGIDs, prefetch );
+}
+
+const brion::SynapseSummary& Circuit::getSynapseSummary() const
+{
+    if( !_impl->synapseSummary )
+        _impl->synapseSummary.reset( new brion::SynapseSummary(
+                           _impl->_synapseSource.getPath() + summaryFilename ));
+    return *_impl->synapseSummary;
+}
+
+const brion::Synapse& Circuit::getSynapseAttributes( const bool afferent ) const
+{
+    const size_t i = afferent ? 0 : 1;
+    if( !_impl->synapseAttributes[i] )
+        _impl->synapseAttributes[i].reset( new brion::Synapse(
+                                               _impl->_synapseSource.getPath() +
+                             (afferent ? afferentFilename : efferentFilename)));
+    return *_impl->synapseAttributes[i];
+}
+
+const brion::Synapse* Circuit::getSynapseExtra() const
```

Throw instead of return 0? It looks a bit asymmetrical otherwise.